packages/plugin-ext/src/main/browser/debug/debug-main.ts

```diff
@@ -284,6 +284,17 @@ export class DebugMainImpl implements DebugMain, Disposable {
         return !!session;
     }
 
+    async $stopDebugging(sessionId?: string): Promise<void> {
+        if (sessionId) {
+            const session = this.sessionManager.getSession(sessionId);
+            return this.sessionManager.terminateSession(session);
+        }
+        // Terminate all sessions if no session is provided.
+        for (const session of this.sessionManager.sessions) {
+            this.sessionManager.terminateSession(session);
```

Comment:
When `$stopDebugging` is called with a `sessionId`, the promise resolves when the session was terminated (or rejected). The promise might resolve earlier when calling the function without the `sessionId` arg. Maybe an `await` is missing from here.

Sorry for the noise if I am wrong 😊